test/libyul/yulOptimizerTests/redundantStoreEliminator/if_overwrite_all_branches.yul

```diff
@@ -0,0 +1,18 @@
+{
+    let c := calldataload(0)
+    // This store will be overwritten in all branches and thus can be removed.
+    sstore(c, 1)
+    if c {
+        sstore(c, 2)
+    }
+    sstore(c, 3)
+}
+// ----
+// step: redundantStoreEliminator
+//
+// {
+//     let c := calldataload(0)
+//     let _2 := 1
+//     if c { let _3 := 2 }
+//     sstore(c, 3)
```

Comment:
Shouldn't this be
```
let _3 = 3
if c { _3 := 2 }
sstore(c, _3)
```
or something like that?

Reply:
No, the first and the second sstore are both overwritten by the third sstore.

Reply:
indeed